### ISSUE TITLE
When I join ssim loss, there is a problem with the depth map

### ISSUE THREAD

**Issue by Samyer** [ClementPinard/SfmLearner-Pytorch#75]
Hi,
When I join ssim loss to the loss function, there is a problem with the depth map. Black bars appear in the depth map. I don't know why this problem occurs.

**Comment by Samyer** [comment]
The result is shown.
![1573462461336](https://user-images.githubusercontent.com/24623502/68574058-b16def00-042e-11ea-9d92-274dd4a2f7e5.jpg)

**Comment by Samyer** [comment]
The result is shown.
![Screenshot from 2019-11-10 16-27-20](https://user-images.githubusercontent.com/24623502/68588273-01e35c00-04c4-11ea-8d77-897cfbe9ee65.png)

**Comment by ClementPinard** [comment]
What do you mean by joing SSIM loss ? Can you be more specific ? Seems like the SSIM dos not behave well in borders, be carfeul with local mean and variance maps, don't pad the image or you will have messed up values at borders , it will think that the point is a mixture of image colors and grey pixels (because it's the default padding value)

**Comment by Samyer** [comment]
Thanks for your suggestions, I found the cause of the problem.


------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;"Clément Pinard"<notifications@github.com&gt;;
发送时间:&nbsp;2019年11月21日(星期四) 晚上8:43
收件人:&nbsp;"ClementPinard/SfmLearner-Pytorch"<SfmLearner-Pytorch@noreply.github.com&gt;;
抄送:&nbsp;"虛·實之間"<847185649@qq.com&gt;;"Author"<author@noreply.github.com&gt;;
主题:&nbsp;Re: [ClementPinard/SfmLearner-Pytorch] When I join ssim loss, there is a problem with the depth map (#75)




What do you mean by joing SSIM loss ? Can you be more specific ? Seems like the SSIM dos not behave well in borders, be carfeul with local mean and variance maps, don't pad the image or you will have messed up values at borders , it will think that the point is a mixture of image colors and grey pixels (because it's the default padding value)
